kani-driver/src/args.rs

```diff
@@ -377,29 +389,47 @@ impl KaniArgs {
         // TODO: these conflicting flags reflect what's necessary to pass current tests unmodified.
         // We should consider improving the error messages slightly in a later pull request.
         if natives_unwind && extra_unwind {
-            Err(Error::with_description(
+            return Err(Error::with_description(
                 "Conflicting flags: unwind flags provided to kani and in --cbmc-args.",
                 ErrorKind::ArgumentConflict,
-            ))
-        } else if self.cbmc_args.contains(&OsString::from("--function")) {
-            Err(Error::with_description(
+            ));
+        }
+        if self.cbmc_args.contains(&OsString::from("--function")) {
+            return Err(Error::with_description(
                 "Invalid flag: --function should be provided to Kani directly, not via --cbmc-args.",
                 ErrorKind::ArgumentConflict,
-            ))
-        } else if self.quiet && self.concrete_playback == Some(ConcretePlaybackMode::Print) {
-            Err(Error::with_description(
+            ));
+        }
+        if self.quiet && self.concrete_playback == Some(ConcretePlaybackMode::Print) {
+            return Err(Error::with_description(
                 "Conflicting options: --concrete-playback=print and --quiet.",
                 ErrorKind::ArgumentConflict,
-            ))
-        } else if self.concrete_playback.is_some() && self.output_format == OutputFormat::Old {
-            Err(Error::with_description(
+            ));
+        }
+        if self.concrete_playback.is_some() && self.output_format == OutputFormat::Old {
+            return Err(Error::with_description(
                 "Conflicting options: --concrete-playback isn't compatible with \
                 --output-format=old.",
                 ErrorKind::ArgumentConflict,
-            ))
-        } else {
-            Ok(())
+            ));
+        }
+        if self.concrete_playback.is_some() && self.jobs() != Some(1) {
+            // Concrete playback currently embeds a lot of assumptions about the order in which harnesses get called.
+            return Err(Error::with_description(
+                "Conflicting options: --concrete-playback isn't compatible with --jobs.",
+                ErrorKind::ArgumentConflict,
+            ));
```

Comment:
Can't these conflicts be specified as an annotation in `KaniArgs` fields? Not clear to me why we handle them here.

Reply:
Only partially. This allows `-j1` to work fine for instance, and is more durable if we end up wanting to change the default to `num_cpus` later, but still need to require single threaded for concrete-playback. (Though hopefully we'll fix that first...)

(Basically this is a semantic check: note the `()` on `self.jobs()`, rather than just a syntactic check that `conflicts` would perform.)